apps/crn-server/test/data-providers/reminders.data-provider.test.ts

```diff
@@ -1146,6 +1147,299 @@ describe('Reminder Data Provider', () => {
       });
     });
 
+    describe('Upload Presentation Reminder', () => {
+      beforeAll(() => {
+        jest.useFakeTimers();
+      });
+
+      afterAll(() => {
+        jest.useRealTimers();
+      });
+
+      describe('When the user is PM of a speakers team and less than 72 hours have passed since the event ended', () => {
+        it.each`
+          asapRole     | teamRole                       | expected
+          ${`Grantee`} | ${`Lead PI (Core Leadership)`} | ${`not fetch`}
+          ${`Grantee`} | ${'Co-PI (Core Leadership)'}   | ${`not fetch`}
+          ${`Grantee`} | ${'Collaborating PI'}          | ${`not fetch`}
+          ${`Grantee`} | ${`Key Personnel`}             | ${`not fetch`}
+          ${`Grantee`} | ${`Scientific Advisory Board`} | ${`not fetch`}
+          ${`Grantee`} | ${`Project Manager`}           | ${`fetch`}
+          ${`Grantee`} | ${`ASAP Staff`}                | ${`not fetch`}
+          ${`Staff`}   | ${`Lead PI (Core Leadership)`} | ${`not fetch`}
+          ${`Staff`}   | ${`Lead PI (Core Leadership)`} | ${`not fetch`}
+          ${`Staff`}   | ${'Co-PI (Core Leadership)'}   | ${`not fetch`}
+          ${`Staff`}   | ${'Collaborating PI'}          | ${`not fetch`}
+          ${`Staff`}   | ${`Key Personnel`}             | ${`not fetch`}
+          ${`Staff`}   | ${`Scientific Advisory Board`} | ${`not fetch`}
+          ${`Staff`}   | ${`Project Manager`}           | ${`not fetch`}
+          ${`Staff`}   | ${`ASAP Staff`}                | ${`not fetch`}
+        `(
+          `Should $expected the reminder when user has asap role $asapRole and team role $teamRole`,
+          async ({ asapRole, teamRole, expected }) => {
+            // set current time to one minute after the end of the fixture event
+            jest.setSystemTime(
+              DateTime.fromISO('2022-01-01T10:01:00Z').toJSDate(),
+            );
+            const squidexGraphqlResponse = getSquidexRemindersGraphqlResponse();
+            squidexGraphqlResponse.queryEventsContents![0]!.flatData.startDate =
+              '2022-01-01T08:00:00Z';
+            squidexGraphqlResponse.queryEventsContents![0]!.flatData.endDate =
+              '2022-01-01T10:00:00Z';
+
+            squidexGraphqlResponse.queryEventsContents![0]!.flatData.speakers![0]!.team![0]! =
+              { id: 'team-id-3' };
+
+            squidexGraphqlResponse.findUsersContent!.flatData.teams = [
+              {
+                id: [{ id: 'team-id-3' }],
+                role: teamRole,
+              },
+            ];
+            squidexGraphqlResponse.findUsersContent!.flatData!.role! = asapRole;
+
+            squidexGraphqlResponse.queryResearchOutputsContents = [];
+            squidexGraphqlClientMock.request.mockResolvedValueOnce(
+              squidexGraphqlResponse,
+            );
+
+            const result = await reminderDataProvider.fetch(
+              fetchRemindersOptions,
+            );
+
+            if (expected === 'fetch') {
```

Comment:
i was never a fan of conditional expectations, maybe we should move these into its own test (as in split the 'fetch' and 'not- fetch' lists)?

Reply:
committed here [remove conditional expect from reminders.data-provider](https://github.com/yldio/asap-hub/pull/2406/commits/910bc19797b94d340fccee72c30d897b0aa33ca1)